kore/src/Kore/Builtin/AssociativeCommutative.hs

```diff
@@ -757,17 +762,155 @@ unifyEqualsNormalized
             HasCallStack =>
             InternalVariable variable =>
             TermLike variable ->
-            MaybeT unifier (TermNormalizedAc normalized variable)
+            unifier (TermNormalizedAc normalized variable)
         normalize1 patt =
             case toNormalized patt of
                 Bottom ->
-                    lift $
-                        Monad.Unify.explainAndReturnBottom
-                            "Duplicated elements in normalization."
-                            first
-                            second
+                    Monad.Unify.explainAndReturnBottom
+                        "Duplicated elements in normalization."
+                        first
+                        second
                 Normalized n -> return n
 
+data UnifyEqualsElementListsData normalized = UnifyEqualsElementListsData
+    { -- Given normalized data norm1, norm2, norm1 - norm2 and norm2 - norm1
+      allElements1, allElements2 :: ![ConcreteOrWithVariable normalized RewritingVariableName]
+    , -- Is Just v if v is the sole opaque in norm1 - norm2, else Nothing
+      maybeVar :: !(Maybe (ElementVariable RewritingVariableName))
+    }
+
+data UnifyEqualsNormAc normalized variable
```

Comment:
I'm not sure what documentation is needed here, since each type and constructor corresponds to an existing unification function of the same (or similar, `UnifyEqualsNormAc` -> `unifyEqualsNormalizedAc`).